macaroonbakery/checkers/namespace.py

```diff
@@ -141,11 +141,23 @@ def deserialize_namespace(data):
     '''
     if isinstance(data, bytes):
         data = data.decode('utf-8')
-    if data == '':
-        return Namespace()
-    kvs = data.split(' ')
+    kvs = data.split()
     uri_to_prefix = {}
     for kv in kvs:
-        k, v = kv.split(':')
-        uri_to_prefix[k] = v
+        i = kv.rfind(':')
+        if i == -1:
+            raise ValueError('no colon in namespace field {}'.format(kv))
+        uri, prefix = kv[0:i], kv[i + 1:]
+        if not is_valid_schema_uri(uri):
+            # Currently this can't happen because the only invalid URIs
+            # are those which contain a space
+            raise ValueError(
+                'invalid URI {} in namespace field {}'.format(uri, kv))
```

Comment:
Perhaps use repr(uri) and repr(kv) (and below) so the messages are more obvious if the strings are empty or otherwise ambiguous?